spec/rubocop/cop/lint/useless_assignment_spec.rb

```diff
@@ -266,6 +266,16 @@ def some_method
         end
       RUBY
     end
+
+    it 'registers offenses for the assignment in numblock' do
+      expect_offense(<<~'RUBY')
+        1.times do
+          puts "Iteration #{_1 + 1}"
+          foo = 2
+          ^^^ Useless assignment to variable - `foo`.
+        end
+      RUBY
```

Comment:
I was able to reproduce it below test case. Can you review and update it?

```suggestion
    it 'registers offenses for the self assignment in numblock', :ruby27 do
      expect_offense(<<~'RUBY')
        do_something { foo += _1 }
                       ^^^ Useless assignment to variable - `foo`. Use `+` instead of `+=`.
      RUBY
```

For reference, the reproduced error is below:

```console
  1) RuboCop::Cop::Lint::UselessAssignment when a block local variable is assigned and unreferenced registers offenses for the self assignment in numblock
     Failure/Error: node.children[child_index]

     TypeError:
       no implicit conversion from nil to integer
     # ./lib/rubocop/cop/variable_force/scope.rb:53:in `body_node'
     # ./lib/rubocop/cop/lint/useless_assignment.rb:99:in `return_value_node_of_scope'
```

Reply:
Oh, I have forgotten to mark the test as :ruby27, that was it! 🤦‍♂️

Reply:
Thank you, @koic! I have amended my commit with the fix.